docs/contribute/testing.adoc

```diff
@@ -0,0 +1,423 @@
+= Writing and running tests for OpenSCAP
+
+This document should help you with writing and running tests for your pull
+requests. It is recommended to add a new test for every new functionality or
+if you are updating the code which is not covered by any test. Another
+recommendation is to add your test in a separate commit.
+
+All the tests reside in the link:../../tests[tests] directory. It has multiple
+subdirectories which should represent various parts of the OpenSCAP library and
+its utilities. When you contribute to some part of the OpenSCAP project you
+should put a test for your contribution into the corresponding subdirectory
+in the link:../../tests[tests] directory.
+
+It would be too time consuming to describe all the parts of the OpenSCAP
+library, its utilities and the test directories corresponding to these parts.
+Therefore, use your best judgement when deciding where to put the test for your
+pull request and if you are not sure don't be affraid to ask in the pull
+request, someone will definitely help you with that.
```

Comment:
so you say here that we are not gonna discuss where to put tests but 2 sections later you describe that :) this might be a leftover

Reply:
Not really, the point of this should be to tell that we don't have a strict rule where to put tests (+ in the current state the test placement is a mess anyway :D ), I will rewrite it to be more clear.